ssl/t1_trce.c

```diff
@@ -1275,6 +1303,37 @@ static int ssl_print_certificate(BIO *bio, int indent,
     return 1;
 }
 
+static int ssl_print_raw_public_key(BIO *bio, const SSL *ssl, int server,
+                                    int indent, const unsigned char **pmsg,
+                                    size_t *pmsglen)
+{
+    EVP_PKEY *pkey;
+    size_t clen;
+    const unsigned char *msg = *pmsg;
+    size_t msglen = *pmsglen;
+
+    if (msglen < 3)
+        return 0;
+    clen = (msg[0] << 16) | (msg[1] << 8) | msg[2];
+    if (msglen < clen + 3)
+        return 0;
+
+    msg += 3;
+
+    BIO_indent(bio, indent, 80);
+    BIO_printf(bio, "raw_public_key, length=%d\n", (int)clen);
+
+    pkey = d2i_PUBKEY_ex(NULL, &msg, clen, ssl->ctx->libctx, NULL);
+    if (pkey == NULL) {
+        return 0;
+    }
+    EVP_PKEY_print_public(bio, pkey, indent + 2, NULL);
```

Comment:
IIRC this dumps hex bytes.  Perhaps a PEM "PUBLIC KEY" would be more useful!

Reply:
This can be tweaked later, if anyone feels sufficiently motivated. Marking resolved.